fix(docs): add a page describing everything about feature statuses (#31941)
mdn/content: files/en-us/mdn/writing_guidelines/page_structures/technology_status/index.md

---
title: Technology status
slug: MDN/Writing_guidelines/Page_structures/Technology_status
page-type: mdn-writing-guide
---

{{MDNSidebar}}

Technology status provides developers with information about the implementation stage of the feature among browser vendors. A technology status is one of the following:

- [`experimental`](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#setting-experimental)
  - : A feature is considered experimental if it is supported by one and only one browser engine. Some exceptions apply, however, for long-standing features and features behind flags and prefixes.
- [`non-standard`](https://github.com/mdn/browser-compat-data/blob/main/schemas/compat-data-schema.md#status-information)
  - : A feature is considered non-standard if it does not exist in any specification.
- [`deprecated`](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#setting-deprecated)
  - : A feature is considered deprecated if it is on the path to removal or the feature is discouraged from use despite its retention for backwards compatibility.
- `standard`

>   - : If none of the above status apply, then the feature is considered standard. We don't need to explicitly mention this status in the content code because the status is assumed by default.

Reviewer: ```suggestion
  - : If none of the above statuses apply, then the feature is considered standard. We don't explicitly add this status to content pages as a technology status is `standard` by default.
```

Reviewer: See later comment

> Technology status provides developers with information about the implementation stage of the feature among browser vendors. A technology status is one of the following:
> 
> - [`experimental`](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#setting-experimental)
>   - : A feature is considered experimental if it is supported by one and only one browser engine. Some exceptions apply, however, for long-standing features and features behind flags and prefixes.
> - [`non-standard`](https://github.com/mdn/browser-compat-data/blob/main/schemas/compat-data-schema.md#status-information)
>   - : A feature is considered non-standard if it does not exist in any specification.
> - [`deprecated`](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#setting-deprecated)
>   - : A feature is considered deprecated if it is on the path to removal or the feature is discouraged from use despite its retention for backwards compatibility.
> - `standard`
>   - : If none of the above status apply, then the feature is considered standard. We don't need to explicitly mention this status in the content code because the status is assumed by default.
> 
> For more information on these terms, see the ["Experimental, deprecated, and obsolete"](/en-US/docs/MDN/Writing_guidelines/Experimental_deprecated_obsolete) page. To see detailed instructions on how to choose status for a feature, refer [choosing status properties](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#choosing-status-properties) section in `@mdn/browser-compat-data` (BCD) repository.

Reviewer: One last comment is that I'm a little concerned that wording drifts in definitions, so I wouldn't define these here, but list and link, what do you think?

```suggestion
A technology status provides developers with information about the implementation stage of the feature among browser vendors and is one of the following:

- [`experimental`](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#setting-experimental)
- [`non-standard`](https://github.com/mdn/browser-compat-data/blob/main/schemas/compat-data-schema.md#status-information)
- [`deprecated`](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#setting-deprecated)

If none of the above statuses apply, then the feature is considered `standard`.
We don't explicitly add this status to content pages as a technology status is `standard` by default.
For more information on these terms, see the ["Experimental, deprecated, and obsolete"](/en-US/docs/MDN/Writing_guidelines/Experimental_deprecated_obsolete) page.

To see detailed instructions on how to choose a status for a feature, refer to the [choosing status properties](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#choosing-status-properties) section in the `@mdn/browser-compat-data` (BCD) repository.
```

## How to add or update technology statuses

> The technology statuses of all the features documented on MDN are defined in its accompanying [@mdn/browser-compat-data](https://github.com/mdn/browser-compat-data) (BCD) repository entry and are _automatically_ updated in the `mdn/content` repository whenever a new [version of BCD is released](https://github.com/mdn/browser-compat-data/releases).```

Reviewer: Markdown  leftover

```suggestion
The technology statuses of all the features documented on MDN are defined in its accompanying [@mdn/browser-compat-data](https://github.com/mdn/browser-compat-data) (BCD) repository entry and are _automatically_ updated in the `mdn/content` repository whenever a new [version of BCD is released](https://github.com/mdn/browser-compat-data/releases).
```

> **Warning:**

> > Do not manually update technology statuses in the `mdn/content` repository. To update a feature's technology status, you need to [submit a pull request](https://github.com/mdn/browser-compat-data/blob/main/docs/contributing.md#updating-the-compat-data) in the BCD repository. After your changes are approved and merged in BCD, an [automated pull request](https://github.com/search?q=repo%3Amdn%2Fcontent+Synchronize+with+BCD&type=pullrequests) updates the statuses in the `mdn/content` repository.

Author: My initial language [was a suggestion](https://github.com/mdn/content/pull/31941/commits/817c86fa06013ee21d489b2a6b1e50746fad04ac#diff-dd0bb6f913e6339c55b69f997a1717c0d5f86f66541102b0cf17f3e2bba4066cR26).\
Do we need a poll or a broader discussion for "whether to make this a strict rule" or "leave it open ended with a warning"?

> > Do not manually update technology statuses in the `mdn/content` repository. To update a feature's technology status, you need to [submit a pull request](https://github.com/mdn/browser-compat-data/blob/main/docs/contributing.md#updating-the-compat-data) in the BCD repository. After your changes are approved and merged in BCD, an [automated pull request](https://github.com/search?q=repo%3Amdn%2Fcontent+Synchronize+with+BCD&type=pullrequests) updates the statuses in the `mdn/content` repository.

Reviewer 2: TLDR: No. Let's leave it as it is. I don't "like" this but it isn't worth arguing about as long as your script is run at regularly and is reliable. Also ignore my other comments if this is all agreed.

Outside your control, but I'd be much happier if the automation was much deeper - sidebars and these macros from metadata/BCD and out of the templates. As it is now we have "these macros are needed and so we're making you aware of them, but actually you don't have to know about them". So quite a lot of things for people to know, that can't be removed, but that are irrelevant to them.

> > Do not manually update technology statuses in the `mdn/content` repository. To update a feature's technology status, you need to [submit a pull request](https://github.com/mdn/browser-compat-data/blob/main/docs/contributing.md#updating-the-compat-data) in the BCD repository. After your changes are approved and merged in BCD, an [automated pull request](https://github.com/search?q=repo%3Amdn%2Fcontent+Synchronize+with+BCD&type=pullrequests) updates the statuses in the `mdn/content` repository.

Reviewer: >I'd be much happier if the automation was much deeper - sidebars and these macros populated from metadata/BCD and never in the content part of the pages.

I agree, this would be great.

> > Do not manually update technology statuses in the `mdn/content` repository. To update a feature's technology status, you need to [submit a pull request](https://github.com/mdn/browser-compat-data/blob/main/docs/contributing.md#updating-the-compat-data) in the BCD repository. After your changes are approved and merged in BCD, an [automated pull request](https://github.com/search?q=repo%3Amdn%2Fcontent+Synchronize+with+BCD&type=pullrequests) updates the statuses in the `mdn/content` repository.

Reviewer 3: > TLDR: No. Let's leave it as it is. I don't "like" this but it isn't worth arguing about as long as your script is run at regularly and is reliable. Also ignore my other comments if this is all agreed.
> 
> Outside your control, but I'd be much happier if the automation was much deeper - sidebars and these macros populated from metadata/BCD and never in the content part of the pages.

Yes, see also https://github.com/mdn/yari/pull/8289#issuecomment-1442594111 or https://github.com/mdn/yari/pull/7227. These bots are a workaround for things that should and fairly easily could be in the platform.

> > Do not manually update technology statuses in the `mdn/content` repository. To update a feature's technology status, you need to [submit a pull request](https://github.com/mdn/browser-compat-data/blob/main/docs/contributing.md#updating-the-compat-data) in the BCD repository. After your changes are approved and merged in BCD, an automated pull request updates the statuses in the `mdn/content` repository.

Reviewer: Is the code for this automation checked in somewhere? If not, should we put it in GH actions? Currently it's not apparent when this automation runs and what rules it has etc., or at least I can't find it from the PRs.

```suggestion
> Do not manually update technology statuses in the `mdn/content` repository. To update a feature's technology status, you need to [submit a pull request](https://github.com/mdn/browser-compat-data/blob/main/docs/contributing.md#updating-the-compat-data) in the BCD repository. After your changes are approved and merged in BCD, an [automated pull request](https://github.com/search?q=repo%3Amdn%2Fcontent+Synchronize+with+BCD&type=pullrequests) updates the statuses in the `mdn/content` repository.
```

Author: > Is the code for this automation checked in somewhere? If not, should we put it in GH actions?

At the moment the bot is hosted in [my personal repo](https://github.com/OnkarRuikar/temp/tree/main/.github/workflows). We are planning to move all the current and future bots in more public place like `https://github.com/openwebdocs/content-automation`.  As the end result of the bots' actions are only PRs so it shouldn't matter where they are publicly hosted.

## How are technology statuses specified in content?

The following sections document mechanisms using which technology statuses are rendered in MDN documents. As mentioned, you should consider these mechanisms read-only as their inclusion in the content is automated.

### Technology status icons in sidebars

To show status icons for a feature listed in sidebars, the [front-matter](/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types/CSS_function_page_template#sect1) `status` property is used:

```yml
---
title: Feature name
status:
  - experimental
  - non-standard
  - deprecated
browser-compat: api.feature
---
```

> When sidebars are rendered, the code looks for the statuses in each page's front-matter.

Reviewer 4: Even though the "read-only" part is mentioned before, should we again reiterate here:
```suggestion
When sidebars are rendered, the code looks for the statuses in each page's front-matter. As mentioned before, these statuses are updated automatically; you do not need to change them manually.
```

> ```yml
> title: Feature name
> ---
> status:
>   - experimental
>   - non-standard
>   - deprecated
> browser-compat: api.feature
> ```

Reviewer: ````suggestion
```yml
---
title: Feature name
status:
  - experimental
  - non-standard
  - deprecated
browser-compat: api.feature
---
```
````

---
title: Technology status
slug: MDN/Writing_guidelines/Page_structures/Technology_status
page-type: mdn-writing-guide
---

{{MDNSidebar}}

> Technology status of a feature tells developers implementation stage of the feature among browsers. Using it, developers can decide if a feature is stable enough to use in production code. Each feature has one of the following statuses:

Reviewer: ```suggestion
Technology status provides developers with information about the implementation stage of the feature among browser vendors. A technology status is one of the following:
```

  - : A feature is considered non-standard if it does not exist in any specification.
- [`deprecated`](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#setting-deprecated)
  - : A feature is considered deprecated if it is on the path to removal or the feature is discouraged from use despite its retention for backwards compatibility.
- `standard`
  - : If none of the above status apply, then the feature is considered standard. We don't need to explicitly mention this status in the content code because the status is assumed by default.

Detailed information about `experimental` and `standard` can be found on ["Experimental, deprecated, and obsolete"](/en-US/docs/MDN/Writing_guidelines/Experimental_deprecated_obsolete) page. To see detailed instructions on how to choose status for a feature, refer [choosing status properties](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#choosing-status-properties) section in `@mdn/browser-compat-data` (BCD) repository.

## How to add or update technology statuses

> Statuses of all the features are being maintained at only one place: [@mdn/browser-compat-data](https://github.com/mdn/browser-compat-data) (BCD) repository. Those statuses are _automatically_ updated in `mdn/content` repository whenever a new [version of BCD is released](https://github.com/mdn/browser-compat-data/releases) (mostly twice a week).

Reviewer: ```suggestion
The technology status for a feature documented on MDN is defined in its accompanying [@mdn/browser-compat-data](https://github.com/mdn/browser-compat-data) (BCD) repository entry. The `status` front-matter keys are _automatically_ updated in the `mdn/content` repository whenever a new [version of BCD is released](https://github.com/mdn/browser-compat-data/releases), which is twice a week at most.
```

Author: ```suggestion
The technology statuses of all the features documented on MDN are defined in its accompanying [@mdn/browser-compat-data](https://github.com/mdn/browser-compat-data) (BCD) repository entry and are _automatically_ updated in the `mdn/content` repository whenever a new [version of BCD is released](https://github.com/mdn/browser-compat-data/releases).```

> As there is an automation in place to synchronize the data, _there is no need to manually update the statuses in the content code_. You need to [submit a pull request](https://github.com/mdn/browser-compat-data/blob/main/docs/contributing.md#updating-the-compat-data) to BCD repository to update a status. After your changes go live in BCD repository the content repo automatically, at next midnight, receives a pull request that updates the statuses in `content` repository.

Reviewer: ```suggestion
> **Warning:**
> Do not manually update technology statuses in the `mdn/content` repository. To update a feature's technology status, you need to [submit a pull request](https://github.com/mdn/browser-compat-data/blob/main/docs/contributing.md#updating-the-compat-data) in the BCD repository. After your changes are approved and merged in BCD, an automated pull request updates the statuses in the `mdn/content` repository.
```

> **Warning:**
> If you manually update statuses in `content` repository without updating BCD repository, then your changes will get reverted on the next BCD release!
>
> If you update statuses in both `content` and BCD repositories, and content changes get merged before BCD changes go live, then the content changes may get reverted till the BCD changes go live.
>

> > Hence, it is best not to manually update technology statuses in the content code.

Reviewer: You could combine with the warning above (or remove this warning, saying "don't do it" is enough, I think).

Author: Looks like there is no need to go into details of how it fails. Straight up saying "don't do it" sounds good. Removing the warning.

## How are technology statuses specified in content?

> Following sections document mechanisms using which technology statuses are placed in content. As mentioned before, it is not required to use these mechanisms manually.

Reviewer: ```suggestion
The following sections document mechanisms using which technology statuses in content. As mentioned, you should consider these mechanisms read-only as their inclusion in the content is automated.
```

Author: ```suggestion
The following sections document mechanisms using which technology statuses are rendered in MDN documents. As mentioned, you should consider these mechanisms read-only as their inclusion in the content is automated.
```
